addons/base_address_extended/models/base_address_extended.py

```diff
@@ -0,0 +1,100 @@
+                                inverse='_set_street', store=True)
+    street_number2 = fields.Char('Door Number', compute='_split_street',
+                                 inverse='_set_street', store=True)
+
+    def get_street_fields(self):
+        """Returns the fields that can be used in a street format.
+        Overwrite this function if you want to add your own fields."""
+        return STREET_FIELDS
+
+    @api.multi
+    def _set_street(self):
+        """Updates the street field.
+        Writes the `street` field on the partners when one of the sub-fields in STREET_FIELDS
+        has been touched"""
+        for partner in self:
+            street_format = (partner.country_id.street_format or
+                '%(street_number)s/%(street_number2)s %(street_name)s')
+            street_vals = {field: partner[field] or '' for field in self.get_street_fields()}
+            partner.street = street_format % street_vals
+
+    @api.multi
+    @api.depends('street', 'country_id.street_format')
+    def _split_street(self):
+        """Splits street value into sub-fields.
+        Recomputes the fields of STREET_FIELDS when `street` of a partner is updated"""
+        street_fields = self.get_street_fields()
+        for partner in self:
+            if not partner.street:
+                partner.street_name = ''
+                partner.street_number = ''
+                partner.street_number2= ''
+                continue
+
+            street_format = (partner.country_id.street_format or
+                '%(street_number)s/%(street_number2)s %(street_name)s')
+            vals = {}
+            previous_pos = 0
+            street_raw = partner.street
+            field_name = None
+            #iter on fields in street_format, detected as '%(<field_name>)s'
+            for re_match in re.finditer(r'%\(\w+\)s', street_format):
+                field_pos = re_match.start()
+                # get the substring between 2 fields, to be used as separator
+                separator = street_format[previous_pos:field_pos]
+                field_value = None
+                if separator and field_name:
+                    #maxsplit set to 1 to unpack only the first element and let the rest untouched
+                    tmp = street_raw.split(separator, 1)
+                    if len(tmp) == 2:
+                        field_value, street_raw = tmp
+                        vals[field_name] = field_value
+                if field_value or not field_name:
+                    # select next field to find (first pass OR field found)
+                    # [2:-2] is used to remove the extra chars '%(' and ')s'
+                    field_name = re_match.group()[2:-2]
+                else:
+                    # value not found: keep looking for the same field
+                    pass
+                if field_name not in street_fields:
+                    raise UserError(_("Unrecognized field %s in street format.") % field_name)
```

Comment:
What about use comma instead `%`?
`raise UserError(_("Unrecognized field %s in street format."), field_name)`